### PR TITLE
feat: deprecate helmet plugin

### LINKED ISSUE #73: Deprecate gatsby-plugin-react-helmet
Deprecate the plugin in favor of the new https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-head/

### PR DESCRIPTION
**Pull request by e0ipso**
BREAKING CHANGE: existing sites using the theme need to migrate from Helmet to [Head API](https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-head/). See the [example site](https://github.com/Lullabot/gatsby-theme-adr/tree/main/packages/example) to see what needs to happen in your site.

Fixes #73
Refs: #73

### CODE REVIEW

**Comment by e0ipso** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr76-0q0coeqhujjqb3rzbf6mza2erwrvwz87.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6398e1df385f4b3c3f56d15a

Visual Diffs:
- https://dashboard.tugboatqa.com/6398e1df385f4b3c3f56d15a/visualdiffs

**Comment by e0ipso** [comment]
image

**Comment by e0ipso** [comment]
image

File: packages/gatsby-theme-adr/src/components/SEO.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import React, { PropsWithChildren } from 'react';
+import { useSiteMetadata } from '../hooks/use-site-metadata';
+
+export type SEOProps = PropsWithChildren<{
+  title?: string;
+  description?: string;
+  pathname?: string;
+}>;
+
+const SEO = ({ title, description, pathname, children }: SEOProps) => {
```
Comment (isholgueras) [inline packages/gatsby-theme-adr/src/components/SEO.tsx:10]: Does SEO have the same purpose than Helmet? 
Shouldn't be better to name this with Helmet? 
Every time that I see "Helmet" in a React application I think about `<title>`, `<meta>` and so forth.

For me, if I'd have this ADR for internal purposes SEO doesn't makes sense. the `<meta name="twitter...">` are not going to be used, but we have here the title, description and image, and can be used.

File: packages/gatsby-theme-adr/src/components/SEO.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    url: `${siteUrl}${pathname || ``}`,
+    twitterUsername,
+  };
+
+  return (
+    <>
+      <title>{seo.title}</title>
+      <meta name="description" content={seo.description} />
+      <meta name="image" content={seo.image} />
+      <meta name="twitter:card" content="summary_large_image" />
```
Comment (isholgueras) [inline packages/gatsby-theme-adr/src/components/SEO.tsx:32]: Does it make sense to add the Open Graph tags too?

**Review by isholgueras — COMMENTED** [review]
With not so much knowledge on Gatsby, I see this PR is replacing correctly gatsby-plugin-react-helmet with the new Gatsby Head API.

For me looks pretty good.

I've just added a comment for a name. Not very important though

File: packages/gatsby-theme-adr/src/components/SEO.tsx (hunk 1 above)
Comment (e0ipso) [reply packages/gatsby-theme-adr/src/components/SEO.tsx:10]: > For me, if I'd have this ADR for internal purposes SEO doesn't makes sense.

I think ADRs are not internal only. We want them promoted and discoverable.

File: packages/gatsby-theme-adr/src/components/SEO.tsx (hunk 2 above)
Comment (e0ipso) [reply packages/gatsby-theme-adr/src/components/SEO.tsx:32]: :thinking: maybe that **is** a rabbit hole not worth entering. I will remove the Twitter specific ones so we don't favor any particular social network.

Comment (e0ipso) [reply packages/gatsby-theme-adr/src/components/SEO.tsx]: Done.

**Comment by e0ipso** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr76-0q0coeqhujjqb3rzbf6mza2erwrvwz87.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6398e1df385f4b3c3f56d15a

Visual Diffs:
- https://dashboard.tugboatqa.com/6398e1df385f4b3c3f56d15a/visualdiffs

File: packages/gatsby-theme-adr/src/components/SEO.tsx (hunk 2 above)
Comment (isholgueras) [reply packages/gatsby-theme-adr/src/components/SEO.tsx]: Nice :)

**Comment by e0ipso** [comment]
BREAKING CHANGE: existing sites using the theme need to migrate from Helmet to Head API. See the example site to see what needs to happen in your site.

Fixes #73
Refs: #73

**Pull request merged**